ArduPlane/GCS_Mavlink.cpp

```diff
@@ -2067,6 +2067,51 @@ void GCS_MAVLINK::handleMessage(mavlink_message_t* msg)
         break;
     }
 
+    case MAVLINK_MSG_ID_SET_POSITION_TARGET_GLOBAL_INT:
+    {
+        mavlink_set_position_target_global_int_t pos_target;
+        mavlink_msg_set_position_target_global_int_decode(msg, &pos_target);
+        const uint16_t alt_mask = 0b0000000000000100; // (alt mask at bit 3)
+            
+        bool msg_valid = true;
+        AP_Mission::Mission_Command cmd = {0};
+
+        if(pos_target.type_mask ^ alt_mask)
+        {
+            cmd.content.location.alt = pos_target.alt * 100;
+            cmd.content.location.flags.relative_alt = false;
+            cmd.content.location.flags.terrain_alt = false;
+            switch(pos_target.coordinate_frame) {
+                case MAV_FRAME_GLOBAL_INT:
+                {
+                    break;
+                }    
+                case MAV_FRAME_GLOBAL_RELATIVE_ALT_INT:
+                {   
+                    cmd.content.location.flags.relative_alt = true;          
+                    break;
+                }
+                case MAV_FRAME_GLOBAL_TERRAIN_ALT_INT:
+                {
+                    cmd.content.location.flags.terrain_alt = true;
+                    break;
+                }
+                default:
+                {
+                    plane.gcs_send_text_fmt(MAV_SEVERITY_WARNING, "Unexepected coordinate frame flag in MAVLINK_MSG_ID_SET_POSTION_TARGET_GLOBAL_INT message.");
```

Comment:
Ideally we need a way to NACK with all the normal MAV_RESULT enums, but COMMAND_ACK isn't really appropriate here,  Maybe we should stop and coordinate with the PX4 team on getting an appropriate NACK in for these messages, as the GCS needs them to automatically handle failures on setting them (as it shouldn't be scanning text strings).